main.go

```diff
@@ -95,28 +106,24 @@ func main() {
 			args.WaitForDockerSeconds -= 5
 		}
 	}
-	
-	username := args.Username
-	password := args.Password
-	
-	if args.UsernameFILE != "" && args.PasswordFILE != "" {
-		contentUser, err := ioutil.ReadFile(args.UsernameFILE)
-		if err != nil {
-	  		log.Fatal(err)
-		}
-		username = string(contentUser)
-		
-		contentPassword, err := ioutil.ReadFile(args.PasswordFILE)
-		if err != nil {
-	  		log.Fatal(err)
-		}
-		
-		password = string(contentPassword)
+
+	if args.Username == nil && args.UsernameFile != nil {
+		args.Username = &args.UsernameFile.Value
 	}
 
-	if (args.Username != "" || args.Password != "") || (args.UsernameFILE != "" || args.PasswordFILE != "") {
+	if args.Password == nil && args.PasswordFile != nil {
+		args.Password = &args.PasswordFile.Value
+	}
+
+	if args.Username != nil || args.Password != nil {
+		errorMsg := "Username AND password are required for authentication"
+		if args.Username == nil || args.Password == nil {
+			log.Fatalf(errorMsg)
+		}
+		username = strings.TrimSpace(*args.Username)
```

Comment:
If you see my comments above, you shouldn't need this section anymore

Reply:
Then no reason to do this either, right? Let's just make it simple. Revert back `string` without pointer and just check `!=""`. You can remove split and trim here since it's up to the user to make sure that's right.